### DISCUSSION TITLE
Negative Rate no return value

### DISCUSSION THREAD

**Discussion by birnstingl** [jomjol/AI-on-the-edge-device#702] [Q&A] [answered]
When my meter runs into a negative rate problem/error:
**Neg. Rate - Read: 149659 - Raw: 149659 - Pre: 149669** 
no value is returned - but that´s what I want to get back!

My Parameters are:
**PreValueUse = True
AllowNegativeRates = false
ErrorMessage = true or false (makes no difference)**

What am I doing wrong?

Thx for the really great project!!!

**Comment by jomjol** [comment]
Did not get the point of your question. If you run into a wrong value (e.g. negative rate), then no value is returned, because it would not be a valid value. If you access via MQTT, then there will be a error message.
What would you expect to be in the value field itself?

**Reply by birnstingl** [reply]
I am checking the value via wasserzehler.html and based on the ErrorMessage parameter description ("Do not show error message in return value - **_in error case, the last valid number will be send out_** ") I expected the last valid number
So this parameter only applies to MQTT access?

**Comment by jomjol** [comment, marked as answer]
Sorry for confusion. The parameter "ErrorMessage" was used in an old concept for writting the error directly to the "value" field. Since there is now a dedicated "error" field, this is not necessary anymore. I will change this in one of the next versions.

Regarding the behaviour: if a false value is detected, it is not send anymore.
